### PR TITLE
Pull EthClient out of store

### PR DESCRIPTION
**Pull request by samsondav**

### CODE REVIEW

File: core/internal/cltest/cltest.go [hunk 1]
```diff
@@ -128,27 +129,6 @@ func init() {
 	gomega.SetDefaultEventuallyTimeout(3 * time.Second)
 	lvl := logLevelFromEnv()
 	logger.SetLogger(logger.CreateTestLogger(lvl))
-	// Register txdb as dialect wrapping postgres
```
Comment (jkongie) [inline core/internal/cltest/cltest.go:131]: Can we remove this yet? Don't we need it for other tests still using the store?

File: core/internal/cltest/factories.go [hunk 2]
```diff
@@ -498,7 +498,7 @@ func MustInsertTaskRun(t *testing.T, store *strpkg.Store) (uuid.UUID, models.Job
 	return taskRunID, jobRun
 }
 
-func NewEthTx(t *testing.T, store *strpkg.Store, fromAddress common.Address) bulletprooftxmanager.EthTx {
+func NewEthTx(t *testing.T, fromAddress common.Address) bulletprooftxmanager.EthTx {
```
Comment (jkongie) [inline core/internal/cltest/factories.go:501]: Nice!

File: go.mod [hunk 3]
```diff
@@ -29,8 +29,7 @@ require (
 	github.com/jackc/pgconn v1.8.1
 	github.com/jackc/pgtype v1.7.0
 	github.com/jackc/pgx/v4 v4.11.1-0.20210326152507-88ede6efb5b0
-	github.com/jinzhu/gorm v1.9.16
```
Comment (jkongie) [inline go.mod:32]: 🥳

File: core/internal/cltest/factories.go [hunk 4]
```diff
@@ -727,7 +736,7 @@ func MustGenerateRandomKey(t testing.TB, opts ...interface{}) ethkey.Key {
 
 	key := ethkey.Key{
 		Address:   eip,
-		JSON:      gormpostgrestypes.Jsonb{RawMessage: keyjsonbytes},
+		JSON:      datatypes.JSON(keyjsonbytes),
```
Comment (connorwstein) [inline core/internal/cltest/factories.go:739]: oh nice - I think earlier I was under the impression the gorm v2 datatypes didn't support jsonb, but looks like they default to that for postgres https://github.com/go-gorm/datatypes/blob/master/json.go#L75

File: core/internal/cltest/cltest.go (hunk 1 above)
Comment (jkongie) [reply core/internal/cltest/cltest.go:131]: Actually I see it's in just moved to pgtest

Comment (samsondav) [reply core/internal/cltest/cltest.go:131]: IT moved to pgtest

**Review by jkongie — APPROVED** [review]

File: core/store/dialects/dialects.go [hunk 5]
```diff
@@ -11,16 +11,7 @@ const (
 	// When the connection is opened, it starts a transaction and all
 	// operations performed on the DB will be within that transaction.
 	//
-	// HACK: This must be the string 'cloudsqlpostgres' because of an absolutely
```
Comment (connorwstein) [inline core/store/dialects/dialects.go:14]: 🥳

File: go.mod (hunk 3 above)
Comment (connorwstein) [inline go.mod:32]: 🥳

**Review by connorwstein — APPROVED** [review]

File: go.mod (hunk 3 above)
Comment (connorwstein) [reply go.mod:32]: hahah I dropped the same emoji

**Pull request merged**